### PR TITLE
Create sushiswap-lp-actions sql

### LINKED ISSUE #57: [Table] SushiSwap Add / Remove Liquidity

### PR DESCRIPTION
**Pull request by D0R11**
This is the branch to create the table for sushiswap-lp-actions. Here's the sample output running in locally:

```root@c1b01df5aae8:/harmony# dbt run --select sushiswap_lp_actions
Running with dbt=0.21.1
Found 37 models, 303 tests, 0 snapshots, 0 analyses, 180 macros, 0 operations, 8 
seed files, 6 sources, 0 exposures

14:24:43 | Concurrency: 4 threads (target='dev')
14:24:43 |
14:24:43 | 1 of 1 START incremental model DEV.sushiswap_lp_actions.............. ....... [RUN]
14:26:20 | 1 of 1 OK created incremental model DEV.sushiswap_lp_actions......... ....... [SUCCESS 1 in 97.00s]
14:26:20 |
14:26:20 | Finished running 1 incremental model in 103.20s.

Completed successfully

Done. PASS=1 WARN=0 ERROR=0 SKIP=0 TOTAL=1
```

### CODE REVIEW

File: models/core/sushiswap_lp_actions.sql [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}}
+
+
+with 
+
+events as (
+    select 
+        log_id,
+        block_timestamp,
+        date_trunc('day', block_timestamp) as block_date,
```
Comment (antonyip) [inline models/core/sushiswap_lp_actions.sql:22]: no need for block_date

File: models/core/sushiswap_lp_actions.sql [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        tx_hash,
+        event_index,
+        native_contract_address,
+        evm_contract_address,
+
```
Comment (antonyip) [inline models/core/sushiswap_lp_actions.sql]: clean up spaces

File: models/core/sushiswap_lp_actions.sql [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                then 'ADD_LIQUIDITY'
+            else 'REMOVE_LIQUIDITY'
+        end as action,
+    
+        event_inputs:amount0::int as amount0_raw,
+    
+        event_inputs:amount1::int as amount1_raw
+    
+    from {{ ref('logs') }}
+    where (event_name='Mint' or
```
Comment (antonyip) [inline models/core/sushiswap_lp_actions.sql:37]: you will need the incremental table limit here.

File: models/core/sushiswap_lp_actions.sql [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    where (event_name='Mint' or 
+           event_name='Burn') 
+        and event_inputs:sender = '0x1b02da8cb0d097eb8d57a175b88c7d8b47997506'
+),
+
+txs as (
+    select
+        tx_hash,
+        from_address
+    from {{ ref('txs') }}
```
Comment (antonyip) [inline models/core/sushiswap_lp_actions.sql:46]: you'll need incremental limit here

File: models/core/sushiswap_lp_actions.sql [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        unique_key='log_id',
+        tags=['core', 'sushiswap_lp_actions'],
+        cluster_by=['log_id']
+        )
+}}
+
+
+with 
+
+events as (
```
Comment (antonyip) [inline models/core/sushiswap_lp_actions.sql:18]: seperate this into 2 CTE's, one to just get logs, then do your subselect in the 2nd.

put the selection CTE's on top. (the one with just select * from logs where incremental_timestamp)

File: models/core/sushiswap_lp_actions.sql [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            on events_liquidity_pools.token0=token0_prices_usd.token_address 
+                and events_liquidity_pools.block_date=token0_prices_usd.block_date
+        -- join token prices table to token1
+        left join token_prices_usd as token1_prices_usd
+            on events_liquidity_pools.token1=token1_prices_usd.token_address 
+                and events_liquidity_pools.block_date=token1_prices_usd.block_date
+    order by events_liquidity_pools.block_timestamp desc
+)
+
+select * from final_table
```
Comment (antonyip) [inline models/core/sushiswap_lp_actions.sql]: add line

File: models/core/sushiswap_lp_actions.sql [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            as amount1_usd
+    
+    from events_liquidity_pools
+        join txs 
+            on events_liquidity_pools.tx_hash=txs.tx_hash
+        -- join token prices table to token0
+        left join token_prices_usd as token0_prices_usd
+            on events_liquidity_pools.token0=token0_prices_usd.token_address 
+                and events_liquidity_pools.block_date=token0_prices_usd.block_date
+        -- join token prices table to token1
```
Comment (antonyip) [inline models/core/sushiswap_lp_actions.sql]: remove unused code

File: models/core/sushiswap_lp_actions.sql [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        events_liquidity_pools.amount0_raw/pow(10, token0_prices_usd.decimals)*token0_prices_usd.usd_price
+            as amount0_usd,
+    
+        events_liquidity_pools.amount1_raw/pow(10, token1_prices_usd.decimals)*token1_prices_usd.usd_price 
+            as amount1_usd
+    
+    from events_liquidity_pools
+        join txs 
+            on events_liquidity_pools.tx_hash=txs.tx_hash
+        -- join token prices table to token0
```
Comment (antonyip) [inline models/core/sushiswap_lp_actions.sql]: remove unused code

**Review by antonyip — COMMENTED** [review]
Looks good for a first pass :P. Will check the execution of the code later.

File: models/core/sushiswap_lp_actions.sql (hunk 1 above)
Comment (D0R11) [reply models/core/sushiswap_lp_actions.sql:22]: Oh okay even if I'm using it to merge to `tokenprices`?

File: models/core/sushiswap_lp_actions.sql (hunk 5 above)
Comment (D0R11) [reply models/core/sushiswap_lp_actions.sql:18]: So I'll make a new CTE with the incremental load filter then use the logs in the events CTE, is that what you mean?

**Comment by D0R11** [comment]
Output for the latest commit:

```
root@b5cb14d9dd34:/harmony# dbt run --select sushiswap_lp_actions
Running with dbt=0.21.1
Found 37 models, 303 tests, 0 snapshots, 0 analyses, 180 macros, 0 operations, 8 seed files, 6 sources, 0 exposures

17:40:45 | Concurrency: 4 threads (target='dev')
17:40:45 |
17:40:45 | 1 of 1 START incremental model DEV.sushiswap_lp_actions.............. [RUN]
17:40:56 | 1 of 1 OK created incremental model DEV.sushiswap_lp_actions......... [SUCCESS 1 in 11.32s]
17:40:56 |
17:40:56 | Finished running 1 incremental model in 18.00s.
```

File: models/core/sushiswap_lp_actions.sql (hunk 5 above)
Comment (antonyip) [reply models/core/sushiswap_lp_actions.sql:18]: yup

File: models/core/sushiswap_lp_actions.sql (hunk 1 above)
Comment (antonyip) [reply models/core/sushiswap_lp_actions.sql:22]: oh okay, that's fine then..

File: models/core/sushiswap_lp_actions.sql (hunk 5 above)
Comment (D0R11) [reply models/core/sushiswap_lp_actions.sql:18]: Applied the changes on the latest commit. Let me know if there are still changes needed. Thanks!

**Comment by antonyip** [comment]
will look at this and try to finalize it this week.

**Review by antonyip — APPROVED** [review]
Follow up ticket to solve for NULL token0_name in sushiswap_lp_actions.

**Pull request merged**